src/coreclr/jit/emitloongarch64.cpp

```diff
@@ -5064,113 +5068,94 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
             assert(REG_R21 != dst->GetRegNum());
             assert(REG_RA != dst->GetRegNum());
 
-            if (dst->GetRegNum() == regOp1)
+            if (dst->OperGet() == GT_ADD)
             {
-                assert(REG_R21 != regOp1);
-                assert(REG_RA != regOp1);
-                saveOperReg1 = REG_R21;
-                saveOperReg2 = regOp2;
-                emitIns_R_R_R(INS_or, attr, REG_R21, regOp1, REG_R0);
+                saveOperReg1 = (dst->GetRegNum() == regOp1) ? regOp2 : regOp1;
             }
-            else if (dst->GetRegNum() == regOp2)
+            else
             {
-                assert(REG_R21 != regOp2);
-                assert(REG_RA != regOp2);
-                saveOperReg1 = regOp1;
-                saveOperReg2 = REG_R21;
-                emitIns_R_R_R(INS_or, attr, REG_R21, regOp2, REG_R0);
+                if (dst->GetRegNum() == regOp1)
+                {
+                    assert(REG_R21 != regOp1);
+                    assert(REG_RA != regOp1);
+                    saveOperReg1 = REG_R21;
+                    emitIns_R_R_R(INS_or, attr, REG_R21, regOp1, REG_R0);
+                }
+                else
+                {
+                    saveOperReg1 = regOp1;
+                }
             }
-            else
+
+            if (!(dst->gtFlags & GTF_UNSIGNED))
```

Comment:
```suggestion
            if ((dst->gtFlags & GTF_UNSIGNED) == 0)
```
(Coding conventions)

Reply:
Thank you! I will pay attention to these in future PRs.